### PR TITLE
Actually call argument-less methods.

### PR DESCRIPTION
**Pull request by Hixie**
I accidentally broke this in an earlier patch.

### CODE REVIEW

File: packages/pigeon/lib/dart_generator.dart [hunk 1]
```diff
@@ -134,16 +135,16 @@ void _writeFlutterApi(
                   : func.returnType == 'void'
                       ? 'return;'
                       : 'return null;';
-              indent.write('if (message == null) ');
-              indent.scoped('{', '}', () {
-                indent.writeln(emptyReturnStatement);
-              });
               String call;
               if (argType == 'void') {
+                indent.writeln('// ignore message');
```
Comment (blasten) [inline packages/pigeon/lib/dart_generator.dart:140]: Is `message` always `null` when `argType` is `void`?

File: packages/pigeon/mock_handler_tester/test/widget_test.dart [hunk 2]
```diff
@@ -46,7 +53,34 @@ void main() {
...
+      StandardMessageCodec(),
+    ).send(null), isEmpty);
+    try {
+      await const BasicMessageChannel<Object>(
+        'dev.flutter.pigeon.Api.search',
+        StandardMessageCodec(),
+      ).send(null) as Map<Object, Object>;
+      expect(true, isFalse); // should not reach here
+    } catch (error) {
+      expect(error, isAssertionError);
```
Comment (blasten) [inline packages/pigeon/mock_handler_tester/test/widget_test.dart:82]: does it make sense to also verify the message?

File: packages/pigeon/test/dart_generator_test.dart [hunk 3]
```diff
@@ -144,7 +144,7 @@ void main() {
     final StringBuffer sink = StringBuffer();
     generateDart(DartOptions(), root, sink);
     final String code = sink.toString();
-    expect(code, isNot(matches('=.*doSomething')));
+    expect(code, isNot(matches('[^!]=.*doSomething')));
```
Comment (blasten) [inline packages/pigeon/test/dart_generator_test.dart:147]: minor nit: some comment would be helpful. It took me a while to figure out what it was checking :)

**Review by blasten — APPROVED** [review]
LGTM!

File: packages/pigeon/lib/dart_generator.dart (hunk 1 above)
Comment (Hixie) [reply packages/pigeon/lib/dart_generator.dart:140]: Do you mean in pigeon-generated code? Or?

File: packages/pigeon/test/dart_generator_test.dart (hunk 3 above)
Comment (Hixie) [reply packages/pigeon/test/dart_generator_test.dart:147]: fair, me too...

File: packages/pigeon/lib/dart_generator.dart (hunk 1 above)
Comment (blasten) [reply packages/pigeon/lib/dart_generator.dart:140]: yup. pigeon-generated code

Comment (Hixie) [reply packages/pigeon/lib/dart_generator.dart:140]: it looks like `_writeHostApi` always sends null if the argType is void, but of course people can always bypass pigeon and send whatever.

File: packages/pigeon/mock_handler_tester/test/widget_test.dart (hunk 2 above)
Comment (Hixie) [reply packages/pigeon/mock_handler_tester/test/widget_test.dart:82]: done

File: packages/pigeon/test/dart_generator_test.dart (hunk 3 above)
Comment (Hixie) [reply packages/pigeon/test/dart_generator_test.dart:147]: done

**Comment by Hixie** [comment]
Blocked on https://github.com/flutter/flutter/pull/74446 which fixes the test failure il `local_tests` here.

**Comment by Hixie** [comment]
Horrah the tests pass now.

**Comment by Hixie** [comment]
Landing!

**Pull request merged**